core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -317,6 +317,9 @@ private[spark] class BlockManager(
 
   /**
    * Put the block locally, using the given storage level.
+   *
+   * '''Important!''' Callers must not mutate or release the data buffer underlying `bytes`. Doing
```

Comment:
I've explicitly documented the fact that callers must not mutate the data buffers underlying `bytes`.